### ISSUE TITLE
what is the role of model.add_loss(customized_loss) - ( customize the regularizer of weights across different layers.)

### ISSUE THREAD

**Issue by zjfheart** [keras-team/keras#10828]
```

from __future__ import absolute_import
from __future__ import division
from __future__ import print_function

from keras.models import Model
from keras.layers import Input

from keras.engine.topology import Layer
from keras.layers.core import  Dense


import six
from keras import backend as K
from keras.utils.generic_utils import serialize_keras_object
from keras.utils.generic_utils import deserialize_keras_object
from keras import regularizers

class Regularizer(object):
    """Regularizer base class.
    """

    def __call__(self, x):
        return 0.

    @classmethod
    def from_config(cls, config):
        return cls(**config)


class My_regularizer(Regularizer):

    def __init__(self):
        pass

    def __call__(self, x):
        w1 = x[0]
        w2 = x[1]
        
        return K.sum ( K.abs( K.identity(w1) - K.identity(w2) )  ) 
        
    def get_config(self):
        return {}

    
class Eig_regularizer(Regularizer):

    def __init__(self):
        #self.l1 = K.cast_to_floatx(l1)
        #self.l2 = K.cast_to_floatx(l2)
        pass

    def __call__(self, x):
        w1 = x[0]
        w2 = x[1]
        w1 = K.eval(w1)
        w2 = K.eval(w2)
        w = np.dot(w1, w2)
        eig_value, eig_vector = LA.eig(w)
        eig_max = np.max( eig_value.real )
        return K.identity(eig_max) * 1.0
        
    def get_config(self):
        return {}

    
class CustomRegularization(Layer):
    def __init__(self, **kwargs):
        super(CustomRegularization, self).__init__(**kwargs)

    def call(self ,x ,mask=None):
        ld=x[0]
        rd=x[1]
        bce = objectives.binary_crossentropy(ld, rd)
        loss2 = K.sum(bce)
        self.add_loss(loss2,x)
        #you can output whatever you need, just update output_shape adequately
        #But this is probably useful
        return bce

    def get_output_shape_for(self, input_shape):
        return (input_shape[0][0],4)

    
    

def res_reg_Block(Input):
    shortcut = Input
    layer1 = layers.Dense(4)
    y = layer1 (Input)
    w_1 = layer1.trainable_weights[0]
    y = layers.ReLU()(y)

    layer2 =  layers.Dense(4)
    y = layer2 (y)
    w_2 = layer2.trainable_weights[0]
    
    y_scaled = layers.Lambda(lambda x: 0.1 *y) (y)
    y = layers.Add()([y_scaled, shortcut])
    
    
    return y, [w_1, w_2]


N = 10
Input = layers.Input(shape=(4,))
W = []
y, w =res_reg_Block(Input)
W.append(w)
for _ in range(0,N-1):
    y, w = res_reg_Block(y)
    W.append(w)

out = keras.layers.Dense(4, activation='softmax', use_bias = True)(y)

model2 = models.Model(inputs = [Input], outputs = out)

model2.compile(
 optimizer = "adam",
 loss = "categorical_crossentropy",
 metrics = ["accuracy"]
)
for i in W:
    eig_reg = Eig_regularizer()(i)
    model2.add_loss( eig_reg )
```

May I know whether I successfully add regularizer in this way?  

Why when I check 

`model.loss`

It returns
`categorical_crossentropy`
